### PR TITLE
CI: Parallel bedrock contracts tests

### PR DESCRIPTION
**Pull request by trianglesphere**
**Description**

Makes them parallel.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -306,7 +307,12 @@ jobs:
           working_directory: packages/contracts-bedrock
       - run:
           name: test and generate coverage
-          command: yarn coverage:lcov
+          command: |
+            forge --version
+            circleci tests glob "contracts/test/*.sol" "contracts/test/*/*.sol" | circleci tests split | tr "\n" "|" | tee splits.txt
+            cat splits.txt | tr "|" "\n"
```
Comment (maurelian) [inline .circleci/config.yml:313]: Is this line just for debugging?

File: .circleci/config.yml [hunk 2]
```diff
@@ -306,7 +307,12 @@ jobs:
...
+            forge test --list --match-path=$(cat splits.txt)
+            yarn coverage:lcov $(cat splits.txt)
```
Comment (maurelian) [inline .circleci/config.yml:315]: This is going to run the tests twice, which is why previously we've just relied on the coverage command to run the tests.

File: .circleci/config.yml (hunk 1 above)
Comment (trianglesphere) [reply .circleci/config.yml:313]: yes

**Review by mslipper — APPROVED** [review]
Looks fine, would get a stamp from @maurelian to be sure we're not missing anything.

**Comment by trianglesphere** [comment]
Closing in favor of https://github.com/ethereum-optimism/optimism/pull/5114

**Pull request closed without merging**